### PR TITLE
nsqadmin: Fix handling of IPv6 broadcast addresses

### LINKED ISSUE #815: Nsqadmin fails to load node data with IPv6 broadcast addresses
When nsqd instances are configured with IPv6 addresses e.g. `broadcast-address=fdeb:5ba3:18b3::1` then when nsqadmin attempts to load the node stats it does a naive mapping of the broadcast address to an HTTP/TCP address which gives invalid results.  In the above case it would attempt to fetch from `http://fdeb:5ba3:18b3::1:4151/stats` which fails at the URI parsing stage.

Note that if you set the broadcast address to `[fdeb:5ba3:18b3::1]` this fixes nsqadmin behaviour but breaks nsq behaviour on the node as the address gets double-bracketed at some point and ends up trying to connect to `[[fdeb:5ba3:18b3::1]]:4150`.

### PR DESCRIPTION
**Pull request by magnetised**
Fixes #815

### CODE REVIEW

File: internal/clusterinfo/types.go [hunk 1]
```diff
@@ -77,11 +78,29 @@ func (p *Producer) Address() string {
 }
 
 func (p *Producer) HTTPAddress() string {
-	return fmt.Sprintf("%s:%d", p.BroadcastAddress, p.HTTPPort)
+	return addressWithPort(p.BroadcastAddress, p.HTTPPort)
```
Comment (mreiferson) [inline internal/clusterinfo/types.go]: go has builtins for this, let's use [`net.JoinHostPort`](https://golang.org/pkg/net/#JoinHostPort)

**Comment by magnetised** [comment]
@mreiferson Done. standard library FTW.

**Comment by mreiferson** [comment]
Re-triggered those flakey test runs. In the meantime would you mind squashing down to one commit?

**Comment by mreiferson** [comment]
apparently I don't know how to use git or github anymore ^^^

**Comment by mreiferson** [comment]
@magnetised thanks!

**Comment by magnetised** [comment]
@mreiferson I'd forgotten all about this - thanks for the unintentional email-storm reminder :)

**Comment by magnetised** [comment]
@mreiferson any idea when the next release is?

**Comment by mreiferson** [comment]
> @mreiferson any idea when the next release is?

Trying to get things in shape for one!

**Comment by magnetised** [comment]
@mreiferson great. good luck

**Pull request closed without merging**